tests/src/JIT/SIMD/Vector3Interop.cs

```diff
@@ -0,0 +1,339 @@
+            Console.WriteLine("    Managed Vector3 v2: ({0} {1} {2})", cdt.vecs.b.X, cdt.vecs.b.Y, cdt.vecs.b.Z);
+            Console.WriteLine("    Managed Vector3 v3: ({0} {1} {2})", cdt.v3.X, cdt.v3.Y, cdt.v3.Z);
+            Console.WriteLine("    Managed string arg: {0}", cdt.str); 
+        
+            // Expected return value = 2 + 3 + 4 + 6 + 7 + 8 + 11 + 12 + 13 = 93
+            float sum = cdt.vecs.a.X + cdt.vecs.a.Y + cdt.vecs.a.Z 
+                + cdt.vecs.b.X + cdt.vecs.b.Y + cdt.vecs.b.Z 
+                + cdt.v3.X + cdt.v3.Y + cdt.v3.Z;
+            if ((sum != 93) || (cdt.iv != 100) || (cdt.str.ToString() != "ret_string") )
+            {
+                Console.WriteLine("PInvoke Vector3InStruct test failed");
+                return false;            
+            }
+        }        
+
+        Console.WriteLine("All PInvoke testcases passed");
+        return true;
+    }    
+}
+
+class RPInvokeTest 
+{
+    public delegate void CallBackDelegate_RPInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
+        Vector3 v2);
+        
+    public delegate Vector3 CallBackDelegate_RPInvoke_Vector3Ret();
+    
+    public delegate void CallBackDelegate_RPInvoke_Vector3Array(
+        [In, MarshalAs(UnmanagedType.LPArray, SizeParamIndex=1)] Vector3[] v, 
+        int size); 
+        
+    public delegate void CallBackDelegate_RPInvoke_Vector3InStruct(
+        DT v);        
+        
+    public delegate void CallBackDelegate_RPInvoke_Vector3InComplexStruct(
+        ref ComplexDT v);    
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_RPInvoke_Vector3Arg(
+        CallBackDelegate_RPInvoke_Vector3Arg callBack);
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern bool nativeCall_RPInvoke_Vector3Ret(
+        CallBackDelegate_RPInvoke_Vector3Ret callBack);        
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_RPInvoke_Vector3Array(
+        CallBackDelegate_RPInvoke_Vector3Array callBack, 
+        int v);
+           
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_RPInvoke_Vector3InStruct(
+        CallBackDelegate_RPInvoke_Vector3InStruct callBack, 
+        int v);
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]        
+    public static extern bool nativeCall_RPInvoke_Vector3InComplexStruct(
+        CallBackDelegate_RPInvoke_Vector3InComplexStruct callBack);
+    
+    static bool result = false;
+   
+    static void callBack_RPInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
```

Comment:
Also please add some comments to each of the test cases to indicate which assumption(s) it is validating.